Library/Homebrew/dev-cmd/edit.rb

```diff
@@ -50,42 +50,56 @@ def edit
         [HOMEBREW_REPOSITORY]
       end
     else
+      edit_api_message_displayed = T.let(false, T::Boolean)
       args.named.to_paths.select do |path|
-        next path if path.exist?
+        core_formula_path = path.fnmatch?("**/homebrew-core/Formula/*.rb", File::FNM_DOTMATCH)
+        core_cask_path = path.fnmatch?("**/homebrew-cask/Casks/*.rb", File::FNM_DOTMATCH)
```

Comment:
`File::FNM_DOTMATCH` allows relative path arguments to be handled correctly.

Reply:
`**/homebrew-core/Formula/**.rb` is what you want.
```
brew(main):020:0> Pathname("/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/firefox.rb").fnmatch?("**/homebrew-cask/Casks/**.rb", File::FNM_DOTMATCH)
=> true
brew(main):021:0> Pathname("/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/f/firefox.rb").fnmatch?("**/homebrew-cask/Casks/**.rb", File::FNM_DOTMATCH)
=> true
```

Reply:
Exactly, thanks.